internal/server/server.go

```diff
@@ -70,14 +87,78 @@ func New(opts ...ServerFuncOpt) (http.Handler, error) {
 
 	// configure server
 	s.cookies = sessions.NewCookieStore([]byte(s.sessionSecret))
+	if err := s.ConfigureOpenID(); err != nil {
+		return nil, err
+	}
 	s.routes()
 
 	return s, nil
 }
 
+func httpClientForRootCAs(rootCABytes []byte) (*http.Client, error) {
+	tlsConfig := tls.Config{RootCAs: x509.NewCertPool()}
+	if !tlsConfig.RootCAs.AppendCertsFromPEM(rootCABytes) {
+		return nil, fmt.Errorf("no certs found in rootCABase64")
+	}
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tlsConfig,
+			Proxy:           http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		},
+	}, nil
+}
+
+func (s *Server) ConfigureOpenID() error {
+	var err error
+
+	// if the client has not been overridden with a mock client
+	if s.client == nil {
+		if s.clusterCA != "" {
+			// decode root certificates
+			rootCABytes, err := base64.StdEncoding.DecodeString(s.clusterCA)
+			if err != nil {
+				return err
+			}
+			// get HTTP Client with custom root CAs
+			s.client, err = httpClientForRootCAs(rootCABytes)
+			if err != nil {
+				return err
+			}
+		} else {
+			s.client = &http.Client{
+				Timeout: 10 * time.Second,
+			}
```

Comment:
nit: This can be a default statement before s.clusterCA is evaluated.

Reply:
Fixed